common-app-lib/src/cmr/common_app/services/search.clj

```diff
@@ -55,17 +77,63 @@
   (fn [context query results]
     [(:concept-type query) (qm/base-result-format query)]))
 
-(defn find-concepts
-  "Executes a search for concepts using the given query."
-  [context concept-type query]
-  (validate-query context query)
+(defn- add-scroll-results-to-cache
+  "Adds the given search results (truncated to ony the :hits, :timed-out, and :scroll-id keys) 
+  and result string to the cache using the scroll-id as the key"
+  [context scroll-id results result-str]
+  (when scroll-id
+    (let [scroll-result-cache (cache/context->cache context scroll-first-page-cache-key)
+          partial-results (select-keys results [:hits :timed-out])]
+      (cache/set-value scroll-result-cache scroll-id [partial-results result-str]))))
+
+(defn- pop-scroll-results-from-cache
+  "Returns the first page of results from a scroll session (along with the original query
```

Comment:
Nit: I don't think the original query execution time is relevant for this call, though I don't know how it's being used. It seems like it should just be however long it takes to construct the response from the cache.